x-pack/legacy/plugins/lens/server/usage/collectors.ts

```diff
@@ -21,25 +21,124 @@ export function registerLensUsageCollector(
       };
     };
     config: KibanaConfig;
+    server: Server;
   }
 ) {
+  let isCollectorReady = false;
+  async function determineIfTaskManagerIsReady() {
+    let isReady = false;
+    try {
+      isReady = await isTaskManagerReady(plugins.server);
+    } catch (err) {} // eslint-disable-line
+
+    if (isReady) {
+      isCollectorReady = true;
+    } else {
+      setTimeout(determineIfTaskManagerIsReady, 500);
+    }
+  }
+  determineIfTaskManagerIsReady();
+
   const lensUsageCollector = plugins.usage.collectorSet.makeUsageCollector({
     type: 'lens',
-    fetch: async (callCluster: CallCluster): Promise<LensUsage> => {
+    fetch: async (): Promise<LensUsage> => {
       try {
-        return getVisualizationCounts(callCluster, plugins.config);
+        const docs = await getLatestTaskState(plugins.server);
+        // get the accumulated state from the recurring task
+        const state: LensTelemetryState = get(docs, '[0].state');
+
+        const dates = Object.keys(state.byDate || {}).map(dateStr => parseInt(dateStr, 10));
+        const suggestionDates = Object.keys(state.suggestionsByDate || {}).map(dateStr =>
+          parseInt(dateStr, 10)
+        );
+
+        const eventsLast30: Record<string, number> = {};
+        const eventsLast90: Record<string, number> = {};
+        const suggestionsLast30: Record<string, number> = {};
+        const suggestionsLast90: Record<string, number> = {};
+
+        const last30 = moment()
```

Comment:
Your dates and suggestionDates looks identical, only for separate collections. Might be worth pulling out into a function.